### PR TITLE
Week 16 - Malin G & Sofia AW

### PR DESCRIPTION
**Pull request by Sofarz**
Link to my deployed project: https://zen-lamarr-f77825.netlify.app/

My reflections on how this week's project turned out: see ReadME

Things I'd like to have clarified or explained in more detail: where and how to implement localStorage in the right way.

### CODE REVIEW

File: code/src/components/GameBoard.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                                    <ActionContainer>
+                                        <ActionDescription>{action.description}</ActionDescription>
+                                        <button 
+                                            type="button" 
+                                            class="nes-btn is-success"
+                                            aria-label='direction button'
+                                            onClick={() => dispatch(continueGame(action.direction, userName))}
+                                        >
+                                            {action.direction}
+                                        </button>
```
Comment (lars-soderman) [inline code/src/components/GameBoard.js:46]: Interesting that you chose to pass the username in the dispatch, we did it in a different way where we got it directly from state. But that was kind of a hassle so this way maybe is a bit smoother 👍

File: code/src/components/GameContainer.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const GameBoard = () => {
+    return(
+        <>
+            <WelcomePage />
+        </>
+    )
+}
+
+export default GameBoard
```
Comment (lars-soderman) [inline code/src/components/GameContainer.js:13]: What is going on here? 🤔 This component has the same name as the component in file GameBoard.js, and only renders one component. We are unsure if it is in use?

File: code/src/components/SpinningLoader.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (lars-soderman) [inline code/src/components/SpinningLoader.js:2]: We see that you have styled component-styling in a separate folder for a few components, but here you have the styling inside the component - maybe would be more consistent to do them all the same way? 
By the way - this loader is BITCHIN'! (that is a good thing) 👍 🚀 🥇

File: code/src/components/SpinningLoader.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    top: 40%;
+`
+
+const rotation = keyframes`
+    from {
+        transform: rotate(0deg);
+    }
+    to {
+        transform: rotate(359deg);
+    }
```
Comment (lars-soderman) [inline code/src/components/SpinningLoader.js:31]: nice 💯

File: code/src/reducers/ui.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit'
```
Comment (lars-soderman) [inline code/src/reducers/ui.js:1]: Super smurt to do separate reducers for the ui. Even though relatively simple logic in this app, it is good practice. We thunk it very nice. 👍 🧠 🤯

**Review by lars-soderman — APPROVED** [review]
Hey Malin and Sofia! 🤖 🐢  You have made a super great game! We enjoyed getting lost in your maze :) Your styling is fantastic, we thought it was really clever the way you incorporated styled components throughout your project. Your code was clean and easy to read, and we enjoyed seeing your take on the project. We look forward to incorporating some of your features in our project in the future 😉 Our only concern is the mysterious GameContainer component which we are not really sure if it is being used. So, good job, keep on rocking, and - Turtle Power! 🐢 
/Lars, Elaine and Felicia

File: code/src/components/EndOfGame.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import "nes.css/css/nes.min.css"
+
+import { EndGamePageContainer, Header, RestartButtonContainer, Credits } from './styled-components/EndOfGameStyling'
```
Comment (KarolCislowski) [inline code/src/components/EndOfGame.js:5]: It is easier to read for everyone in this way:
```jsx
  import {
    EndGamePageContainer, 
    Header, 
    RestartButtonContainer, 
    Credits} from '../styling/style'
```

File: code/src/components/EndOfGame.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+const EndOfGame = () => {
+    const currentDescription = useSelector((store) => store.game.gameStatus)
+
+    return (
+        <EndGamePageContainer>
```
Comment (KarolCislowski) [inline code/src/components/EndOfGame.js:11]: Indentation should be set to the 2 spaces.

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
After I click the start button I saw this page: 
image
I'm really hungry for adventure so please fix it and let me play ⚔️

File: code/src/components/EndOfGame.js (hunk 6 above)
Comment (Sofarz) [reply code/src/components/EndOfGame.js:5]: If you use a unique username you should be able to play the game 👍 . It is not possible to use the same username twice. :)

**Review by KarolCislowski — APPROVED** [review]
After a deep analysis of your code, I found out that the issue I pointed before was closed by API, not by error inside your code. I sent a memo about it to teachers and I need to sorry for any inconvenience done because of it 😢

**Pull request closed without merging**